database/src/main/postgres/runs/_validate_partitioning.sql

```diff
@@ -0,0 +1,130 @@
+--      i_is_pattern        - flag signaling whether the partitioning is not a real one, but just a pattern.
+--                            If the input partitioning is just a pattern, then values can have NULL values and
+--                            such check would be skipped.
+--
+-- Returns:
+--      Set of issues described as individual textual sentences detailing each problem.
+--
+-------------------------------------------------------------------------------
+DECLARE
+    _mandatory_fields_in_input CONSTANT TEXT[] := ARRAY['keys', 'version', 'keysToValuesMap'];
+    _all_fields_in_input TEXT[];
+
+    _is_input_properly_structured BOOL;
+
+    _partitioning_keys_all_cnt INTEGER;
+    _partitioning_keys_uniq_and_not_null TEXT[];
+    _partitioning_keys_uniq_and_not_null_cnt INTEGER;
+
+    _partitioning_keys_from_values_map TEXT[];
+
+    _partitioning_null_values_cnt INTEGER;
+
+BEGIN
+    -- Checking whether the input has correct structure.
+    SELECT ARRAY(SELECT jsonb_object_keys(i_partitioning))
+    INTO _all_fields_in_input;
+
+    SELECT i_partitioning ?& _mandatory_fields_in_input
+    INTO _is_input_properly_structured;
+    IF NOT _is_input_properly_structured THEN
+        RETURN NEXT
+            'The input partitioning is not properly structured, it should have this structure: '
+                || _mandatory_fields_in_input::TEXT
+                || ' but contains: '
+                || _all_fields_in_input::TEXT;
+    END IF;
+
+    -- Checking whether the array 'keys' is valid, i.e. has unique, non-null elements.
+    SELECT jsonb_array_length(i_partitioning->'keys')
+    INTO _partitioning_keys_all_cnt;
+
+    SELECT array_unique(x.keys)
+    FROM jsonb_to_record(i_partitioning) AS x(keys text[])
+    INTO _partitioning_keys_uniq_and_not_null;
+
+    _partitioning_keys_uniq_and_not_null_cnt := array_length(_partitioning_keys_uniq_and_not_null, 1);
+
+    IF _partitioning_keys_all_cnt != _partitioning_keys_uniq_and_not_null_cnt THEN
+        RETURN NEXT 'The input partitioning is invalid, the keys must be unique and can not contain NULLs: '
+            || (i_partitioning->>'keys');
+    END IF;
+
+    -- Checking whether the map 'keysToValuesMap' has the same keys as the 'keys' attribute.
+    SELECT ARRAY(SELECT jsonb_object_keys(i_partitioning->'keysToValuesMap'))
+    INTO _partitioning_keys_from_values_map;
+
+    IF NOT (
+            (_partitioning_keys_from_values_map @> _partitioning_keys_uniq_and_not_null)
+            AND (_partitioning_keys_from_values_map <@ _partitioning_keys_uniq_and_not_null)
+        ) THEN
+        RETURN NEXT
+            'The input partitioning is invalid, the keys in ''keys'' and ''keysToValuesMap'' do not correspond. '
+                || 'Given in ''keysToValuesMap'': '
+                || _partitioning_keys_from_values_map::TEXT
+                || ' vs (probably expected from ''keys''): '
+                || _partitioning_keys_uniq_and_not_null::TEXT;
+    END IF;
```

Comment:
Shouldn't this be executed only in case of strict check? 🤔

Reply:
will be changed, PG is behaving slightly differently than imperative languages and I respect the way it is generally and in here; I didn't know that retrieving something non-existing in JSONB will yield 'null' instead of some sort of error :)

Reply:
changed